backend/api/src/main/kotlin/software/shonk/application/service/ShorkService.kt

```diff
@@ -70,6 +71,33 @@ class ShorkService(private val shork: IShork) : ShorkUseCase {
         return Result.success(lobby.getStatus())
     }
 
+    override fun getAllLobbies(): AllLobbies {
+        val lobbiesStatus = mutableListOf<Result<Status>>()
+        var lobbyId = 0L
+        while (true) {
+            val currentLobbyStatus = getLobbyStatus(lobbyId)
+            if (currentLobbyStatus.getOrNull() == null) {
+                break
+            }
+            lobbiesStatus.add(currentLobbyStatus)
+            lobbyId++
+        }
+        val lobbies =
+            lobbiesStatus.mapNotNull(Result<Status>::getOrNull).mapIndexed { index, status ->
+                AllLobbies.lobbies(
+                    lobbyId = index,
+                    playersJoined =
+                        listOfNotNull(
+                                if (status.playerASubmitted) "playerA" else null,
+                                if (status.playerBSubmitted) "playerB" else null,
+                            )
+                            .ifEmpty { listOf("no player submission yet") },
+                    gameState = status.gameState.toString(),
+                )
+            }
```

Comment:
All of this would be much better as an iterator over the lobbies.
**Please DO NOT** iterate over all possible integers to find the lobbies that exist.
The information about which lobbies exist already exists in the same exact class.